lib/eyeshade/last_settlement_balance.rb

```diff
@@ -0,0 +1,45 @@
+module Eyeshade
+  class LastSettlementBalance < BaseBalance
+    attr_reader :date, :amount_settlement_currency, :timestamp, :settlement_currency
+
+    def initialize(rates, default_currency, transactions)
+      super(rates, default_currency)
+
+      last_settlement = calculate_last_settlement(transactions)
+      if last_settlement.present?
+        @amount_bat = last_settlement["amount_bat"]
+        @timestamp = last_settlement["timestamp"]
+        @settlement_currency = last_settlement["currency"]
+        @amount_settlement_currency = convert(@amount_bat, @settlement_currency)
+      end
+    end
+
+    private
+
+    def calculate_last_settlement(transactions)
+      return {} if transactions == []
+       # Find most recent settlement transaction
+      last_transaction = transactions.select { |transaction|
+        transaction["settlement_amount"].present?
+      }.last # Eyeshade returns transactions ordered_by created_at
+
+      return {} if last_transaction.nil?
```

Comment:
If transactions is an array this won't ever be nil. Instead you should use `last_transactions.blank?` so that if last_transaction is nil or empty then this will return. 

You also can remove `return {} if transactions == []`